src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -283,7 +286,7 @@ internal async Task<bool> ContainsSymbolsWithNameAsync(Func<string, bool> predic
 
         internal async Task<IEnumerable<Document>> GetDocumentsWithNameAsync(Func<string, bool> predicate, SymbolFilter filter, CancellationToken cancellationToken)
         {
-            return (await _solution.State.GetDocumentsWithNameAsync(Id, predicate, filter, cancellationToken).ConfigureAwait(false)).Select(s => _solution.GetDocument(s.Id));
+            return (await _solution.State.GetDocumentsWithNameAsync(Id, predicate, filter, cancellationToken).ConfigureAwait(false)).Select(s => _solution.GetDocument(s.Id)!);
```

Comment:
💡 Consider a single read of `_solution` to clarify validity of `!`

Reply:
_solution is readonly, what would that help with?